typescript-conditionals/main.ts

```diff
@@ -0,0 +1,157 @@
+/* exported isUnderFive,
+            isEven,
+            startsWithJ,
+            isOldEnoughToDrink,
+            isOldEnoughToDrive,
+            isOldEnoughToDrinkAndDrive,
+            categorizeAcidity,
+            introduceWarnerBro,
+            recommendMovie
+ */
+function isUnderFive(num: number): boolean {
+  return num < 5;
+}
+
+function isEven(num: number): boolean {
+  if (num % 2 === 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function startsWithJ(str: string): boolean {
+  const firstChar = str[0];
+  switch (firstChar) {
+    case 'J':
+      return true;
+    default:
+      return false;
+  }
+}
+
+interface Person {
+  name: string;
+  age: number;
+}
+
+function isOldEnoughToDrink(person: Person): boolean {
+  return person.age >= 21;
+}
+
+function isOldEnoughToDrive(person: Person): boolean {
+  return person.age >= 16;
+}
+
+function isOldEnoughToDrinkAndDrive(person: Person): boolean {
+  const theNum = 5;
+  if (theNum === 5 && typeof person !== 'string') {
+    return false;
+  }
+  return false;
+}
```

Comment:
is a person old enough to drink or old enough to drive if a variable number is 5 and the type of the parameter is not a string?